clinica/iotools/converters/adni_to_bids/adni_utils.py

```diff
@@ -254,12 +264,50 @@ def center_nifti_origin(input_image, output_image):
             qform[i - 1, 3] = -1.0 * hd['pixdim'][i] * hd['dim'][i] / 2.0
         new_img = nib.Nifti1Image(canonical_img.get_data(caching='unchanged'), qform)
 
-    except (OSError, nib.orientations.OrientationError) as e:
-        return None
+        nib.save(new_img, output_image)
+        if not isfile(output_image):
+            error_str = Fore.RED + '[Error] Nifti created but clinica could not save it to ' \
+                        + output_image + '. Please check that the output folder has the correct permissions.' \
+                        + Fore.RESET
+    return output_image, error_str
 
-    nib.save(new_img, output_image)
 
-    return output_image
+def center_all_nifti(bids_dir, output_dir):
+    from colorama import Fore
+    from clinica.utils.io import check_bids_folder
+    from os.path import join
+    from glob import glob
+    from os import listdir
+    from os.path import isdir, isfile
+    from shutil import copy2, copytree
+
+    # output and input must be different, so that we do not mess with user's data
+    if bids_dir == output_dir:
+        # TODO replace ValueError by ClinicaBIDSError once custom clinica error exception are merged in dev
+        raise ValueError(Fore.RED + '[Error] output and output directories must be different' + Fore.RESET)
+
+    # check that input is a BIDS dir
+    check_bids_folder(bids_dir)
+
+    for f in listdir(bids_dir):
+        if isdir(join(bids_dir, f)) and not isdir(join(output_dir, f)):
+            copytree(join(bids_dir, f), join(output_dir, f))
+        elif isfile(join(bids_dir, f)) and not isfile(join(output_dir, f)):
+            copy2(join(bids_dir, f), output_dir)
+
+    pattern = join(output_dir, '**/*.nii*')
+    nifti_files = glob(pattern, recursive=True)
+    all_errors = []
+    for f in nifti_files:
+        print('Handling ' + f)
+        _, current_error = center_nifti_origin(f, f)
+        if current_error:
+            all_errors.append(current_error)
+    if len(all_errors) > 0:
+        final_error_msg = Fore.RED + '[Error] Clinica encoutered ' + str(len(all_errors)) + ' while trying to center all Niftis.\n'
```

Comment:
Did you miss str(len(all_errors)) + " error(s) while..." ?

Reply:
Corrected